query/src/sql/exec/mod.rs

```diff
@@ -364,6 +366,12 @@ impl PipelineBuilder {
             })?;
         }
 
+        // Since transform has been added, making group expressions as column expr is safe.
+        group_expressions
+            .iter_mut()
+            .filter(|expr| matches!(expr, Expression::Column(_)).not())
+            .for_each(|expr| *expr = Expression::Column(expr.column_name()));
+
```

Comment:
Maybe we can use a `iter().filter().map().collect()` instead?

Reply:
If the original group expression is a column, then it will not be in the collect result.